metabase/metabase
# Data app navigation

Resolves #25011, part of #24861

Implements basic data app navigation. When opening a data app from Metabase navigation sidebar, you'll now see a "Launch app" button at the top right. Once you click a button, the app is going to take over the whole Metabase navigation and the sidebar is going to now show a list of app pages (dashboards).

### To Verify

1. Follow steps from #24944 to create an app
2. Save/move a few dashboards into it (apps are collections _technically_)
3. Click the app name in Metabase nav sidebar
4. Ensure the URL is now `/apps/:id-:slug`
5. Click the "Launch app" button at the top right
6. Ensure you now see a list of dashboards in the sidebar instead of regular navigation bar
7. Ensure the URL is now `/a/:id-:slug`
8. Click on a dashboard in the app navigation bar
9. Ensure you can see the dashboard and the URL is now `/a/:appId/page/:dashboardId`
10. Click "Exit app", ensure you're back to /apps/:id-:slug` and the usual navigation sidebar is back

### Demo

https://user-images.githubusercontent.com/17258145/186692613-7c5fbce7-5da8-4725-98db-61473c9499da.mp4
